### PR TITLE
PIN 1799

### PR DESCRIPTION
**Pull request by TonioGela**
This Pr contains some fixes to the comments on https://github.com/pagopa/interop-be-backend-for-frontend/pull/37

### CODE REVIEW

File: src/main/scala/it/pagopa/interop/backendforfrontend/service/TenantManagementService.scala [hunk 1]
```diff
@@ -10,7 +10,7 @@ trait TenantManagementService {
 
   def getTenant(tenantId: UUID)(implicit contexts: Seq[(String, String)]): Future[Tenant]
 
-  def getBySelfcareId(getBySelfcareId: UUID)(implicit contexts: Seq[(String, String)]): Future[Tenant]
+  def getBySelfcareId(getBySelfcareId: String)(implicit contexts: Seq[(String, String)]): Future[Tenant]
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/backendforfrontend/service/TenantManagementService.scala]:
```suggestion
  def getBySelfcareId(selfcareId: String)(implicit contexts: Seq[(String, String)]): Future[Tenant]
```

File: src/test/scala/it/pagopa/interop/backendforfrontend/AuthorizationApiServiceSpec.scala [hunk 2]
```diff
@@ -134,19 +134,19 @@ class AuthorizationApiServiceSpec extends AnyWordSpecLike with SpecHelper with S
...
       (mockPartyProcess
-        .getInstitution(_: UUID)(_: Seq[(String, String)], _: ExecutionContext))
+        .getInstitution(_: String)(_: Seq[(String, String)], _: ExecutionContext))
         .expects(selfcareId, *, *)
         .once()
         .returns(
           Future.successful(
             Institution(
-              id = selfcareId,
+              id = UUID.fromString(selfcareId),
```
Comment (galales) [inline src/test/scala/it/pagopa/interop/backendforfrontend/AuthorizationApiServiceSpec.scala:149]: how is it possible this is not breaking? 🤔

File: src/test/scala/it/pagopa/interop/backendforfrontend/AuthorizationApiServiceSpec.scala [hunk 3]
```diff
@@ -32,9 +32,9 @@ class AuthorizationApiServiceSpec extends AnyWordSpecLike with SpecHelper with S
 
     "succeed when the tenant is present" in {
 
-      val uid: String      = UUID.randomUUID().toString
-      val selfcareId: UUID = UUID.randomUUID()
-      val tenantId: UUID   = UUID.randomUUID()
+      val uid: String        = UUID.randomUUID().toString
+      val selfcareId: String = "stringThatWillMakeGalalaesHappy"
```
Comment (galales) [inline src/test/scala/it/pagopa/interop/backendforfrontend/AuthorizationApiServiceSpec.scala]: pls 🙂

**Review by galales — COMMENTED** [review]
the test should break

File: src/test/scala/it/pagopa/interop/backendforfrontend/AuthorizationApiServiceSpec.scala (hunk 2 above)
Comment (TonioGela) [reply src/test/scala/it/pagopa/interop/backendforfrontend/AuthorizationApiServiceSpec.scala:149]: We never released the interop-party-whatever project with the signature update probably

Comment (galales) [reply src/test/scala/it/pagopa/interop/backendforfrontend/AuthorizationApiServiceSpec.scala:149]: sorry I didn't explain the point properly: as far as I see, `selfcareId` is a string not formatted as UUID, so should't `UUID.fromString` throw an exception when running the test?

Comment (TonioGela) [reply src/test/scala/it/pagopa/interop/backendforfrontend/AuthorizationApiServiceSpec.scala:149]: Well, it breaks
```
- should succeed even if the tenant needs to be upserted *** FAILED ***
[info]   java.lang.IllegalArgumentException: Invalid UUID string: stringThatWillMakeGalalaesHappy
[info]   at java.base/java.util.UUID.fromString(UUID.java:215)
[info]   at it.pagopa.interop.backendforfrontend.AuthorizationApiServiceSpec.$anonfun$new$4(AuthorizationApiServiceSpec.scala:149)```

**Review by galales — APPROVED** [review]

**Review by beetlecrunch — APPROVED** [review]

**Pull request merged**